bridge/sam/pool.go

```diff
@@ -0,0 +1,399 @@
+package sam
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type pool struct {
+	// write-lock is called only when changing validators process
+	// otherwise read-lock is called
+	// Changing validators will occur the most rarely (once per epoch)
```

Comment:
We can rephrase this to `write-lock is called only during the validator change process`.

Also, we can rephrase this part as well to `Changing validators occurs rarely (once per epoch)`

Reply:
Fixed at c92a67bf10bc73bcdf8929dca389a8e53020bf04